packages/flutter_tools/lib/src/build_system/targets/localizations.dart

```diff
@@ -0,0 +1,61 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_tools/src/artifacts.dart';
+import 'package:flutter_tools/src/base/io.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+
+import '../../base/file_system.dart';
+import '../build_system.dart';
+
+Future<void> generateLocalizations({
+  @required String flutterRoot,
+  @required FileSystem fileSystem,
+  @required ProcessManager processManager,
+  @required Artifacts artifacts,
+  @required Logger logger,
+}) async {
+  final String genL10nPath = fileSystem.path.join(
+    flutterRoot,
+    'dev',
+    'tools',
+    'localization',
+    'bin',
+    'gen_l10n.dart',
+  );
+  final ProcessResult result = await processManager.run(<String>[
```

Comment:
If I'm understanding this correctly, we need to provide this script with a list of the localization files. It also supports customizing the localization class and output file.

@shihaohong do you anticipate that users would pass these via the command line?

Alternatively we could just look for `arb` files in a known directory